provider/oci/environ.go

```diff
@@ -393,10 +403,12 @@ func (e *Environ) DestroyController(ctx envcontext.ProviderCallContext, controll
 
 	err = e.StopInstances(ctx, ids...)
 	if err != nil {
-		return err
+		err = providerCommon.HandleCredentialError(err, ctx)
+		return errors.Trace(err)
 	}
 	logger.Debugf("Cleaning up network resources")
-	return e.cleanupNetworksAndSubnets(controllerUUID, "")
+	err = e.cleanupNetworksAndSubnets(controllerUUID, "")
+	return providerCommon.HandleCredentialError(err, ctx)
```

Comment:
This spot does it "correctly", in accordance with that the providerCommon.HandleCredentialError() method signature offers.